diff --git a/node_modules/react-native/scripts/react_native_pods.rb b/node_modules/react-native/scripts/react_native_pods.rb
index 1b8a078..d9c2101 100644
--- a/node_modules/react-native/scripts/react_native_pods.rb
+++ b/node_modules/react-native/scripts/react_native_pods.rb
@@ -110,7 +110,8 @@ def use_flipper!(versions = {}, configurations: ['Debug'])
   pod 'FlipperKit/FlipperKitHighlightOverlay', versions['Flipper'], :configurations => configurations
   pod 'FlipperKit/FlipperKitLayoutTextSearchable', versions['Flipper'], :configurations => configurations
   pod 'FlipperKit/FlipperKitNetworkPlugin', versions['Flipper'], :configurations => configurations
-  pod 'OpenSSL-Universal', versions['OpenSSL-Universal'], :configurations => configurations
+  # We need OpenSSL-Universal also in Release build for the Passport NFC
+  # pod 'OpenSSL-Universal', versions['OpenSSL-Universal'], :configurations => configurations
 end
 
 def has_pod(installer, name)
