diff --git a/lib/module/ReanimatedModule/js-reanimated/JSReanimated.js b/lib/module/ReanimatedModule/js-reanimated/JSReanimated.js
index 7f3193536638d962805d80869847ac5dc0c0252f..c20b1f4006ef1758629d66231fe2d901abb440d1 100644
--- a/lib/module/ReanimatedModule/js-reanimated/JSReanimated.js
+++ b/lib/module/ReanimatedModule/js-reanimated/JSReanimated.js
@@ -209,7 +209,6 @@ class JSReanimated {
     throw new Error('[Reanimated] setViewStyle is not available in JSReanimated.');
   }
   setCSSEventHandler(_handler) {
-    throw new Error('[Reanimated] `setCSSEventHandler` is not available in JSReanimated.');
   }
   markNodeAsRemovable(_shadowNodeWrapper) {
     throw new Error('[Reanimated] markNodeAsRemovable is not available in JSReanimated.');
diff --git a/src/ReanimatedModule/js-reanimated/JSReanimated.ts b/src/ReanimatedModule/js-reanimated/JSReanimated.ts
index e53b3e1db10f68bb1fd0582a5f0a4f966908b1dc..130d8a2455d3bd5b0a938dcd4bbf4d44d513da0a 100644
--- a/src/ReanimatedModule/js-reanimated/JSReanimated.ts
+++ b/src/ReanimatedModule/js-reanimated/JSReanimated.ts
@@ -283,9 +283,6 @@ class JSReanimated implements IReanimatedModule {
   }
 
   setCSSEventHandler(_handler: CSSEventHandler): void {
-    throw new Error(
-      '[Reanimated] `setCSSEventHandler` is not available in JSReanimated.'
-    );
   }
 
   markNodeAsRemovable(_shadowNodeWrapper: ShadowNodeWrapper): void {
