diff --git a/android/build.gradle b/android/build.gradle
index 1c2da1a58336d7c856e7ccb964e346ea13c2760c..8ba3fdeff00ab3902225e5cea8311fd5a4b63792 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -35,17 +35,9 @@ def resolveReactNativeDirectory() {
     )
 }
 
-def isNewArchitectureEnabled() {
-  return rootProject.hasProperty("newArchEnabled") && rootProject.getProperty("newArchEnabled") == "true"
-}
-
 apply plugin: 'com.android.library'
 apply plugin: 'maven-publish'
 
-if (isNewArchitectureEnabled()) {
-  apply plugin: "com.facebook.react"
-}
-
 def getExtOrDefault(name) {
   return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties["UltimateConfig_" + name]
 }
@@ -68,7 +60,6 @@ android {
   defaultConfig {
     minSdkVersion getExtOrIntegerDefault("minSdkVersion")
     targetSdkVersion getExtOrIntegerDefault("targetSdkVersion")
-    buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
   }
   buildTypes {
     release {
@@ -104,14 +95,6 @@ dependencies {
   }
 }
 
-if (isNewArchitectureEnabled()) {
-  react {
-    jsRootDir = file("../src/")
-    libraryName = "UltimateConfig"
-    codegenJavaPackageName = "com.reactnativeultimateconfig"
-  }
-}
-
 afterEvaluate { project ->
   task androidSourcesJar(type: Jar) {
     archiveClassifier = 'sources'
@@ -124,7 +107,7 @@ afterEvaluate { project ->
     def javaCompileTask = variant.javaCompileProvider.get()
 
     task "jar${name}"(type: Jar, dependsOn: javaCompileTask) {
-      from javaCompileTask.destinationDir
+      from javaCompileTask.destinationDirectory
     }
   }
 
diff --git a/index.js b/index.js
index 13cbc93bfd56eb74c4f6e49a0f15d353c0a21023..deffba1b81725eae49ba2eec8ca66540528081ea 100644
--- a/index.js
+++ b/index.js
@@ -3,4 +3,4 @@ import override from "./override";
 
 const { UltimateConfig } = NativeModules;
 
-export default { ...UltimateConfig, ...override };
+export default { ...UltimateConfig, ...UltimateConfig.getConstants(), ...override };
diff --git a/ios/UltimateConfig.h b/ios/UltimateConfig.h
index 41ad9efb55a3fd96986a49ef9e369e5b6271e328..492c89f81a637c65b3749b0a4b37f8ca17cfb78f 100644
--- a/ios/UltimateConfig.h
+++ b/ios/UltimateConfig.h
@@ -1,12 +1,6 @@
 
-#ifdef RCT_NEW_ARCH_ENABLED
-#import "RNUltimateConfigSpec.h"
-
-@interface UltimateConfig : NSObject <NativeUltimateConfigSpec>
-#else
 #import <React/RCTBridgeModule.h>
 
 @interface UltimateConfig : NSObject <RCTBridgeModule>
-#endif
 
 @end
diff --git a/ios/UltimateConfig.m b/ios/UltimateConfig.m
index 7debfdff0402385a71111c72517799ec25128e8d..c76bfdb218f014ef643971fa308c85a4133c0bb6 100644
--- a/ios/UltimateConfig.m
+++ b/ios/UltimateConfig.m
@@ -14,13 +14,4 @@ - (NSDictionary *)constantsToExport
     return getValues();
 }
 
-// Don't compile this code when we build for the old architecture.
-#ifdef RCT_NEW_ARCH_ENABLED
-- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:
-    (const facebook::react::ObjCTurboModule::InitParams &)params
-{
-    return std::make_shared<facebook::react::NativeUltimateConfigSpecJSI>(params);
-}
-#endif
-
 @end
diff --git a/react-native-ultimate-config.podspec b/react-native-ultimate-config.podspec
index 3421407f6b9230bcbd91377bcb5248a8d51d5171..5fd0bb32c5f962668fe630147d9bc2176a3ed5ff 100644
--- a/react-native-ultimate-config.podspec
+++ b/react-native-ultimate-config.podspec
@@ -1,7 +1,6 @@
 require "json"
 
 package = JSON.parse(File.read(File.join(__dir__, "package.json")))
-folly_compiler_flags = '-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1 -Wno-comma -Wno-shorten-64-to-32'
 
 Pod::Spec.new do |s|
   s.name         = "react-native-ultimate-config"
@@ -20,19 +19,4 @@ Pod::Spec.new do |s|
   s.source_files = "ios/**/*.{h,m,mm}"
 
   s.dependency "React-Core"
-
-  # Don't install the dependencies when we run `pod install` in the old architecture.
-  if ENV['RCT_NEW_ARCH_ENABLED'] == '1' then
-    s.compiler_flags = folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED=1"
-    s.pod_target_xcconfig    = {
-        "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\"",
-        "OTHER_CPLUSPLUSFLAGS" => "-DFOLLY_NO_CONFIG -DFOLLY_MOBILE=1 -DFOLLY_USE_LIBCPP=1",
-        "CLANG_CXX_LANGUAGE_STANDARD" => "c++17"
-    }
-    s.dependency "React-Codegen"
-    s.dependency "RCT-Folly"
-    s.dependency "RCTRequired"
-    s.dependency "RCTTypeSafety"
-    s.dependency "ReactCommon/turbomodule/core"
-  end
 end
