diff --git a/android/src/main/java/com/screenshotprevent/ScreenshotPreventModule.kt b/android/src/main/java/com/screenshotprevent/ScreenshotPreventModule.kt
index 4f702213610a7c8b8746204c8873a4c6f71b7223..b41f729bba5109249f1b041286c0470d9f1f9ec9 100644
--- a/android/src/main/java/com/screenshotprevent/ScreenshotPreventModule.kt
+++ b/android/src/main/java/com/screenshotprevent/ScreenshotPreventModule.kt
@@ -22,7 +22,7 @@ class ScreenshotPreventModule internal constructor(context: ReactApplicationCont
 
   private var lifecycleEventListener = object : LifecycleEventListener {
     override fun onHostResume() {
-      currentActivity?.let {
+      reactApplicationContext.getCurrentActivity()?.let {
         if (overlayLayout != null) {
           it.runOnUiThread {
             (it.window.decorView.rootView as ViewGroup).removeView(overlayLayout)
@@ -39,7 +39,7 @@ class ScreenshotPreventModule internal constructor(context: ReactApplicationCont
     }
 
     override fun onHostPause() {
-      currentActivity?.let {
+      reactApplicationContext.getCurrentActivity()?.let {
         if (overlayLayout != null) {
           it.runOnUiThread {
             val layoutParams = RelativeLayout.LayoutParams(
@@ -71,7 +71,7 @@ class ScreenshotPreventModule internal constructor(context: ReactApplicationCont
 
   @ReactMethod
   override fun enabled(enabled: Boolean) {
-    currentActivity?.let {
+    reactApplicationContext.getCurrentActivity()?.let {
       it.runOnUiThread {
         if (enabled) {
           it.window.setFlags(
@@ -87,7 +87,7 @@ class ScreenshotPreventModule internal constructor(context: ReactApplicationCont
 
   @ReactMethod
   override fun enableSecureView(imagePath: String?) {
-    currentActivity?.let {
+    reactApplicationContext.getCurrentActivity()?.let {
       if (overlayLayout == null) {
         createOverlayLayout(it, imagePath)
       }
@@ -103,7 +103,7 @@ class ScreenshotPreventModule internal constructor(context: ReactApplicationCont
 
   @ReactMethod
   override fun disableSecureView() {
-    currentActivity?.let {
+    reactApplicationContext.getCurrentActivity()?.let {
       it.runOnUiThread {
         if (overlayLayout != null) {
           (it.window.decorView.rootView as ViewGroup).removeView(overlayLayout)
diff --git a/ios/ScreenshotPrevent.h b/ios/ScreenshotPrevent.h
index 7ac565114db96b9606ade38931c8e31122d3a276..f3f9fda433cbcad4c02f4eb7ae2ffa179a9ef137 100644
--- a/ios/ScreenshotPrevent.h
+++ b/ios/ScreenshotPrevent.h
@@ -1,8 +1,8 @@
 
 #ifdef RCT_NEW_ARCH_ENABLED
 #import "RNScreenshotPreventSpec.h"
-#import "RCTBridgeModule.h"
-#import "RCTConvert.h"
+#import <React/RCTBridgeModule.h>
+#import <React/RCTConvert.h>
 
 @interface ScreenshotPrevent : NSObject <NativeScreenshotPreventSpec>
 #else
diff --git a/ios/ScreenshotPrevent.mm b/ios/ScreenshotPrevent.mm
index 87eb48535f92072c373ab4ccc75969bfc2687546..76680598c91bf239a3f946729d7ed0e6ba1d8721 100644
--- a/ios/ScreenshotPrevent.mm
+++ b/ios/ScreenshotPrevent.mm
@@ -104,59 +104,43 @@ CGSize CGSizeAspectFill(const CGSize aspectRatio, const CGSize minimumSize)
 
 
 /**
- * creates secure text field inside rootView of the app
- * taken from https://stackoverflow.com/questions/18680028/prevent-screen-capture-in-an-ios-app
- *
- * converted to ObjC and modified to get it working with RCT
+ * Creates secure text field inside the UIApplication's `keyWindow`
  */
--(void) addSecureTextFieldToView:(UIView *) view :(NSString *) imagePath {
-
-    UIView *rootView = [UIApplication sharedApplication].keyWindow.rootViewController.view;
-
-
-    // fixes safe-area
-    secureField = [[UITextField alloc] initWithFrame:rootView.frame];
-    secureField.secureTextEntry = TRUE;
-    secureField.userInteractionEnabled = FALSE;
-
-    if (imagePath && ![imagePath isEqualToString:@""]) {
-        UIView * imgView = [[UIView alloc] initWithFrame:CGRectMake(0.f, 0.f, rootView.frame.size.width, rootView.frame.size.height)];
-        NSURL *url = [NSURL URLWithString:imagePath];
-        NSData *data = [NSData dataWithContentsOfURL:url];
-        UIImage *img = [[UIImage alloc] initWithData:data];
-
-        CGSize sizeBeingScaledTo = CGSizeAspectFill(img.size, imgView.frame.size);
-
-        // redraw the image to fit the screen size
-        UIGraphicsBeginImageContextWithOptions(imgView.frame.size, NO, 0.f);
-
-        float offsetX = (imgView.frame.size.width - sizeBeingScaledTo.width) / 2;
-        float offsety = (imgView.frame.size.height - sizeBeingScaledTo.height) / 2;
-
-
-        [img drawInRect:CGRectMake(offsetX, offsety, sizeBeingScaledTo.width, sizeBeingScaledTo.height)];
-        UIImage * resultImage = UIGraphicsGetImageFromCurrentImageContext();
-        UIGraphicsEndImageContext();
+-(void) addSecureView {
+    if (@available(iOS 13.0, *)) {
+        if (secureField == nil) {
+            [self initTextField];
+        }
+        [secureField setSecureTextEntry: TRUE];
+        [secureField setBackgroundColor: UIColor.whiteColor];
+    }
+}
 
-        secureField.backgroundColor = [UIColor colorWithPatternImage:resultImage];
+- (void) initTextField {
+    CGRect screenRect = [[UIScreen mainScreen] bounds];
+    secureField = [[UITextField alloc] initWithFrame:CGRectMake(0, 0, screenRect.size.width, screenRect.size.height)];
+    secureField.translatesAutoresizingMaskIntoConstraints = NO;
+        
+    [secureField setTextAlignment:NSTextAlignmentCenter];
+    [secureField setUserInteractionEnabled: NO];
+        
+    UIWindow *window = [UIApplication sharedApplication].keyWindow;
+    [window.layer.superlayer addSublayer: secureField.layer];
+        
+    if (secureField.layer.sublayers.firstObject) {
+        [secureField.layer.sublayers.firstObject addSublayer: window.layer];
     }
+}
 
-    [view sendSubviewToBack:secureField];
-    [view addSubview:secureField];
-    [view.layer.superlayer addSublayer:secureField.layer];
-    [[secureField.layer.sublayers lastObject] addSublayer:view.layer];
-}
-
-// TODO: not working now, fix crash on _UITextFieldCanvasView contenttViewInvalidated: unrecognized selector sent to instance
--(void) removeSecureTextFieldFromView:(UIView *) view {
-    for(UITextField *subview in view.subviews){
-        if([subview isMemberOfClass:[UITextField class]]) {
-            if(subview.secureTextEntry == TRUE) {
-                [subview removeFromSuperview];
-                subview.secureTextEntry = FALSE;
-                secureField.userInteractionEnabled = TRUE;
-            }
-        }
+- (void) removeSecureView {
+    if (secureField == nil) { return; }
+        
+    UIWindow *window = [UIApplication sharedApplication].keyWindow;
+        
+    [secureField setSecureTextEntry: FALSE];
+    CALayer *textFieldLayer = secureField.layer.sublayers.firstObject;
+    if ([window.layer.superlayer.sublayers containsObject:textFieldLayer]) {
+        [textFieldLayer removeFromSuperlayer];
     }
 }
 
@@ -168,21 +152,12 @@ -(void) removeSecureTextFieldFromView:(UIView *) view {
 
 /** adds secure textfield view */
 RCT_EXPORT_METHOD(enableSecureView: (NSString *)imagePath) {
-    if(secureField.secureTextEntry == false) {
-        UIView *view = [UIApplication sharedApplication].keyWindow.rootViewController.view;
-        for(UIView *subview in view.subviews) {
-            [self addSecureTextFieldToView:subview :imagePath];
-        }
-    }
+    [self addSecureView];
 }
 
 /** removes secure textfield from the view */
 RCT_EXPORT_METHOD(disableSecureView) {
-    secureField.secureTextEntry = false;
-    UIView *view = [UIApplication sharedApplication].keyWindow.rootViewController.view;
-    for(UIView *subview in view.subviews) {
-        [self removeSecureTextFieldFromView:subview];
-    }
+    [self removeSecureView];
 }
 
 // Don't compile this code when we build for the old architecture.
